server/historian/packages/historian-base/src/routes/utils.ts

```diff
@@ -204,47 +160,26 @@ export function queryParamToString(value: any): string {
 	return value;
 }
 
-/**
- * Validate specific request parameters to prevent directory traversal.
- * TODO: replace with validateRequestParams from service-shared.
- */
-export function validateRequestParams(...paramNames: (string | number)[]): RequestHandler {
-	return (req, res, next) => {
-		for (const paramName of paramNames) {
-			const param = req.params[paramName];
-			if (!param) {
-				continue;
-			}
-			if (containsPathTraversal(param)) {
-				return handleResponse(
-					Promise.reject(new NetworkError(400, `Invalid ${paramName}: ${param}`)),
-					res,
-				);
-			}
-		}
-		next();
-	};
-}
-
-export function verifyTokenNotRevoked(
-	revokedTokenChecker: IRevokedTokenChecker | undefined,
-): RequestHandler {
+export function verifyToken(revokedTokenChecker: IRevokedTokenChecker | undefined): RequestHandler {
 	return async (request, response, next) => {
 		try {
-			if (revokedTokenChecker) {
-				const tenantId = request.params.tenantId;
-				const authorization = request.get("Authorization");
-				const token = parseToken(tenantId, authorization);
-				const claims = decode(token) as ITokenClaims;
-
-				let isTokenRevoked = false;
-				if (claims.jti) {
-					isTokenRevoked = await revokedTokenChecker.isTokenRevoked(
-						tenantId,
-						claims.documentId,
-						claims.jti,
-					);
-				}
+			const reqTenantId = request.params.tenantId;
+			const authorization = request.get("Authorization");
+			const token = parseToken(reqTenantId, authorization);
+			const claims = validateTokenClaims(token, "documentId", reqTenantId, false);
+			const documentId = claims.documentId;
+			const tenantId = claims.tenantId;
+			if (containsPathTraversal(documentId)) {
+				// Prevent attempted directory traversal.
+				throw new NetworkError(400, `Invalid document id: ${documentId}`);
+			}
+			// Verify token not revoked if JTI claim is present
+			if (revokedTokenChecker && claims.jti) {
```

Comment:
Maybe worth creating a work item about using verifyToken in server/routerlicious/packages/services-utils/src/auth.ts in this function